src/utils.js

```diff
@@ -21,6 +21,15 @@ export function eachPromise(arr, iterator) {
     )
   , Promise.resolve());
 }
+export function eachPromiseRemove(arr, iterator) {
```

Comment:
it seems that instead of defining this helper function we can just pass a reversed array to `eachPromise`?